project/src/js/ui/tournament/list/tournamentListView.js

```diff
@@ -0,0 +1,76 @@
+import h from '../../helper';
+import {header, empty, pad} from '../../shared/common';
+import layout from '../../layout';
+import i18n from '../../../i18n';
+import m from 'mithril';
+import tabs from 'polythene/tabs/tabs';
+
+export default function view(ctrl) {
+  const bodyCtrl = tournamentListBody.bind(undefined, ctrl);
+
+  return layout.free(header.bind(undefined, i18n('tournaments')), bodyCtrl, empty, empty);
+}
+
+const TABS = [{
+    id: 'started',
+    label: 'In Progress'
+}, {
+    id: 'created',
+    label: 'Upcoming'
+}, {
+    id: 'finished',
+    label: 'Completed'
+}];
+
+function tabNavigation (currentTabFn) {
+    return m('.nav-header', m.component(tabs, {
+        buttons: TABS,
+        autofit: true,
+        selectedTab: currentTabFn(),
+        activeSelected: true,
+        getState: (state) => {
+            currentTabFn(state.index);
+        }
+    }));
+}
+
+function tournamentListBody(ctrl) {
+  if (!ctrl.tournaments()) return null;
```

Comment:
ctrl.tournaments() returns an object with three properties ('started',
'created', 'finished'), each of which is an array, so I think this works as
written.
Check out curl -H "Accept:application/vnd.lichess.v1+json" "
http://en.lichess.org/tournament"

On Fri, Feb 19, 2016 at 8:16 AM, Vincent Velociter <notifications@github.com
> wrote:

> In project/src/js/ui/tournament/list/tournamentListView.js
> <https://github.com/veloce/lichobile/pull/281#discussion_r53458980>:
>
> > +}];
> > +
> > +function tabNavigation (currentTabFn) {
> > +    return m('.nav-header', m.component(tabs, {
> > +        buttons: TABS,
> > +        autofit: true,
> > +        selectedTab: currentTabFn(),
> > +        activeSelected: true,
> > +        getState: (state) => {
> > +            currentTabFn(state.index);
> > +        }
> > +    }));
> > +}
> > +
> > +function tournamentListBody(ctrl) {
> > +  if (!ctrl.tournaments()) return null;
>
> Here ctrl.tournaments() returns an array. Since an empty array evaluates
> to true you need to write this instead:
>
> if (ctrl.tournaments().length === 0) return null;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/veloce/lichobile/pull/281/files#r53458980>.
>